src/Tests/dotnet-new.Tests/DotnetClassTemplateTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Extensions.Logging;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.TemplateEngine.Authoring.TemplateVerifier;
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Cli.New.IntegrationTests
+{
+    public class DotnetClassTemplateTests : BaseIntegrationTest, IClassFixture<SharedHomeDirectory>
+    {
+        private readonly SharedHomeDirectory _fixture;
+        private readonly ITestOutputHelper _log;
+        private readonly ILogger _logger;
+
+        public DotnetClassTemplateTests(SharedHomeDirectory fixture, ITestOutputHelper log) : base(log)
+        {
+            _fixture = fixture;
+            _log = log;
+            _logger = new XunitLoggerProvider(log)
+                .CreateLogger("TestRun");
+        }
+
+        [Theory]
```

Comment:
It would be good to add the tests also for `net6.0` and net8.0` and more options for lang version. Not necessary full set, but maybe starting 7.0 or so.